### DISCUSSION TITLE
[SOLVED] Cleaner syntax or sugar for running tasks (instead of `task |> Async.AwaitTask |> Async.RunSynchronously`)

### DISCUSSION THREAD

**Discussion by drk-mtr** [fsharp/fslang-suggestions#1206] [General]
[EDIT: All irrelevant, please see comment from `theprash` below]

I find myself using `task` more than `async` because of interop with existing dotnet libraries. I sometimes feel that code like this is more verbose than it needs to be.

```fsharp
someTask
|> Async.AwaitTask
|> Async.RunSynchronously
```

I appreciate it would be very easy for me to simply create a function or operator that makes this slightly more succinct. E.g.

```fsharp
let RunTaskSynchronously t = t |> Async.AwaitTask |> Async.RunSynchronously
```

I also appreciate that if I have code like this in lots of different places in my app, I'm almost definitely doing something wrong.

However, does anyone think there's any value in trying to come up with a more succinct standard approach for this? It seems quite verbose for something that comes up often.

There's probably a good reason for this not being cleaner. Obviously we need to keep Async.AwaitTask |> Async.RunSynchronously, so any addition would mean multiple ways to achieve the same thing which is bad. Unfortunately I have no clever ideas on ways around this.

**Comment by vzarytovskii** [comment]
Task builder should probably solve it, or a simple alias to `Async.AwaitTask |> Async.RunSynchronously`

**Comment by theprash** [comment]
You can do `.Result` directly on the task with a similar effect.

**Reply by drk-mtr** [reply]
Thank you very much! I feel very silly now ha.

@vzarytovskii sorry for clogging up the page with my beginner suggestions, please feel free to delete this thread.

**Reply by theprash** [reply]
You're welcome. Don't overuse it though! 😂

**Reply by vzarytovskii** [reply]
> Thank you very much! I feel very silly now ha.
> 
> @vzarytovskii Vlad Zarytovskii FTE sorry for clogging up the page with my beginner suggestions, please feel free to delete this thread.

No worries whatsoever, feel free to ask questions here, or in `dotnet/fsharp`'s discussions!
